FMS/Pages/Facilities/Index.cshtml.cs

```diff
@@ -1,75 +1,112 @@
-        public SelectList FacilityStatuses { get; private set; }
-        public SelectList FacilityTypes { get; private set; }
-        public SelectList BudgetCodes { get; private set; }
-        public SelectList OrganizationalUnits { get; private set; }
-        public SelectList ComplianceOfficers { get; private set; }
-
-        public IndexModel(
-            IFacilityRepository repository,
-            ISelectListHelper listHelper)
-        {
-            _repository = repository;
-            _listHelper = listHelper;
-        }
-
-        public async Task<IActionResult> OnGetAsync()
-        {
-            Spec = new FacilitySpec();
-            await PopulateSelectsAsync();
-            return Page();
-        }
-
-        public async Task<IActionResult> OnGetSearchAsync(FacilitySpec spec, [FromQuery] int p = 1)
-        {
-            // Get the list of facilities matching the "Spec" criteria
-            FacilityList = await _repository.GetFacilityPaginatedListAsync(spec, p, GlobalConstants.PageSize);
-            Spec = spec;
-            ShowResults = true;
-            await PopulateSelectsAsync();
-            return Page();
-        }
-
-        public IActionResult OnPost(FacilitySpec spec)
-        {
-            return RedirectToPage("../Reports/Index", spec);
-        }
-
-        private async Task PopulateSelectsAsync()
-        {
-            BudgetCodes = await _listHelper.BudgetCodesSelectListAsync();
-            ComplianceOfficers = await _listHelper.ComplianceOfficersSelectListAsync();
-            FacilityStatuses = await _listHelper.FacilityStatusesSelectListAsync();
-            FacilityTypes = await _listHelper.FacilityTypesSelectListAsync();
-            OrganizationalUnits = await _listHelper.OrganizationalUnitsSelectListAsync();
-        }
-    }
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using CsvHelper.Configuration;
+using FMS.Domain.Data;
+using FMS.Domain.Dto;
+using FMS.Domain.Dto.PaginatedList;
+using FMS.Domain.Repositories;
+using FMS.Platform.Extensions;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.Mvc.Rendering;
+
+namespace FMS.Pages.Facilities
+{
+    public class IndexModel : PageModel
+    {
+        private readonly IFacilityRepository _repository;
+        private readonly ISelectListHelper _listHelper;
+
+        // "Spec" is the Facility DTO bound to the HTML Page elements
+        public FacilitySpec Spec { get; set; }
+
+        // "ReportSpec" is the Facility DTO Bound to Export Results to Excel (csv file)
+        [BindProperty]
+        public FacilitySpec ReportSpec { get; set; }
+
+        // List of facilities resulting from the search
+        public IPaginatedResult FacilityList { get; private set; }
+
+        // Detailed Facility List to go to a report
+        public IReadOnlyList<FacilityDetailDto> FacilityReportList { get; private set; }

```

Comment:
Since this property is only used in the `OnPostAsync` method, move it there as a local variable instead.